# **Proyek Retail Customer Analytics & Sales Dashboard**

# PHASE 1 - Business Understanding

## 1. Konteks Bisnis
Perusahaan adalah UK-based online retail yang menjual gift-ware secara non-store. Mayoritas pelanggan adalah wholesaler yang melakukan pembelian dalam jumlah besar. Perusahaan beroperasi secara internasional dengan basis pelanggan yang tersebar di berbagai negara.
## 2. Stakeholder
- C-Level/Management - membutuhkan visibility terhadap performa bisnis secara keseluruhan
- Tim Marketing - membutuhkan segmentasi pelanggan untuk strategi retensi dan targeting
## 3. Masalah Bisnis
- Masalah 1: *"Siapa pelanggan paling valuable kami, dan bagaimana cara mempertahankan mereka?"*
- Masalah 2: *"Bagaimana tren performa penjualan kami, dan di mana peluang untuk meningkatkannya?"*
## 4. Pertanyaan Analitik
## Customer Analytics *(Masalah 1)*
- Siapa pelanggan paling valuable berdasarkan RFM?
- Bagaimana segmentasi pelanggan berdasarkan perilaku pembelian?
- Seberapa baik perusahaan mempertahankan pelanggan dari waktu ke waktu?
## Sales Analytics *(Masalah 2)*
- Bagaimana tren revenue bulanan dan tahunan?
- Negara mana yang menghasilkan revenue terbesar?
- Produk apa yang paling laris dan paling sering dibatalkan?
- Berapa Average Order Value (AOV) dan bagaimana tren-nya?
## 5. KPI yang akan Diukur
| KPI | Stakeholder |
|---|---|
| Total Revenue | Management |
| Monthly Revenue Growth (MoM) | Management |
| Average Order Value (AOV) | Management |
| Monthly Revenue Growth (MoM) | Management & Marketing |
| Cancellation Rate | Management |
| Customer Retention Rate | Marketing |
| RFM Segment Distribution | Marketing |
| Cohort Retention Rate | Marketing |


# PHASE 2 - Data Collection & Setup

## Langkah 1 - Struktur Folder Proyek
Buat folder proyek kamu dengan struktur berikut. Kamu bisa buat manual atau lewat terminal VS Code:

```
mkdir customer-sales-analytics
cd customer-sales-analytics
mkdir -p data/raw data/cleaned notebooks dashboard reports
```

Lalu pindahkan file dataset kamu ke folder `data/raw/`:
```
data/raw/online_retail_II.csv
```

Struktur folder sekarang harusnya terlihat seperti ini:
```
customer-sales-analytics/
│
├── 📁 data/
│   ├── raw/
│   │   └── online_retail_II.csv   ← taruh di sini
│   └── cleaned/
│
├── 📁 notebooks/
├── 📁 dashboard/
├── 📁 reports/
└── README.md
```

## Langkah 2 - Library yang Dibutuhkan
Adapun library yang digunakan untuk proyek ini ada dibawah:
```
pip install pandas numpy matplotlib seaborn plotly scikit-learn openpyxl nbformat
```

Agar proyek reproducible, maka perlu dibuat file requirements.txt:
```
pip freeze > requirements.txt
```

## Langkah 3 - Notebook & Load Dataset
Buat notebook `01_data_cleaning.ipynb` untuk proyek ini di folder notebooks, lalu jalankan beberapa kode berikut:

In [1]:
# Import library
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

In [2]:
df = pd.read_csv('../data/raw/online_retail_II.csv')
print(f"Shape: {df.shape}")
df.head()

Shape: (1067371, 8)


,Invoice,StockCode,Description,Quantity,InvoiceDate,Price,Customer ID,Country
0,489434,85048,15CM CHRISTMAS GLASS BALL 20 LIGHTS,12,2009-12-01 07:45:00,6.95,13085.0,United Kingdom
1,489434,79323P,PINK CHERRY LIGHTS,12,2009-12-01 07:45:00,6.75,13085.0,United Kingdom
2,489434,79323W,WHITE CHERRY LIGHTS,12,2009-12-01 07:45:00,6.75,13085.0,United Kingdom
3,489434,22041,"RECORD FRAME 7"" SINGLE SIZE",48,2009-12-01 07:45:00,2.10,13085.0,United Kingdom
4,489434,21232,STRAWBERRY CERAMIC TRINKET BOX,24,2009-12-01 07:45:00,1.25,13085.0,United Kingdom


In [3]:
# Cek data type dan missing value jika ada
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1067371 entries, 0 to 1067370
Data columns (total 8 columns):
 #   Column       Non-Null Count    Dtype  
---  ------       --------------    -----  
 0   Invoice      1067371 non-null  str    
 1   StockCode    1067371 non-null  str    
 2   Description  1062989 non-null  str    
 3   Quantity     1067371 non-null  int64  
 4   InvoiceDate  1067371 non-null  str    
 5   Price        1067371 non-null  float64
 6   Customer ID  824364 non-null   float64
 7   Country      1067371 non-null  str    
dtypes: float64(2), int64(1), str(5)
memory usage: 65.1 MB


In [4]:
# Cek statistik dasar
df.describe()

,Quantity,Price,Customer ID
count,1.067371e+06,1.067371e+06,824364.000000
mean,9.938898e+00,4.649388e+00,15324.638504
std,1.727058e+02,1.235531e+02,1697.464450
min,-8.099500e+04,-5.359436e+04,12346.000000
25%,1.000000e+00,1.250000e+00,13975.000000
50%,3.000000e+00,2.100000e+00,15255.000000
75%,1.000000e+01,4.150000e+00,16797.000000
max,8.099500e+04,3.897000e+04,18287.000000


In [5]:
# Cek jumlah missing value
print(df.isnull().sum())
print(f"\nPersentase missing values:")
print(round(df.isnull().sum() / len(df) * 100, 2))

Invoice             0
StockCode           0
Description      4382
Quantity            0
InvoiceDate         0
Price               0
Customer ID    243007
Country             0
dtype: int64

Persentase missing values:
Invoice         0.00
StockCode       0.00
Description     0.41
Quantity        0.00
InvoiceDate     0.00
Price           0.00
Customer ID    22.77
Country         0.00
dtype: float64


# PHASE 3 - Data Cleaning & Preprocessing
Untuk fase ini kita akan membuat notebook sendiri, yaitu `01_data_cleaning.ipynb`. Ringkasan langkah dan output akan diberikan disini.

## Kondisi Awal
- Total baris: 1.067.371 | Total kolom: 8
- Missing `Customer ID`: 243.007 baris (22.8%)
- Missing `Description`: 4.382 baris (0.4%)
- Duplikat: 34.335 baris (3.2%)
- Cancelled (prefix 'C'): 19.494 baris
- Quantity negatif: 22.950 baris
- Price = 0 / negatif: 6.207 baris

## Aksi yang Dilakukan
1. Pisahkan transaksi cancelled → `cancelled_transactions.csv`
2. Pisahkan transaksi retur → `return_transactions.csv`
3. Drop missing `Customer ID` & `Description`
4. Drop duplikat
5. Drop `Price` = 0 & negatif
6. Convert `InvoiceDate` → datetime
7. Convert `Customer ID` → string
8. Tambah kolom baru: `TotalPrice`, `Year`, `Month`, `DayOfWeek`

## Kondisi Akhir
- Total baris: 779.425 | Total kolom: 12
- Missing values: 0
- Output: `cleaned_data.csv`, `cancelled_transactions.csv`, `return_transactions.csv`, `retail.db`

In [ ]:
# Load dataset yang sudah dibersihkan

df = pd.read_csv('../data/cleaned/cleaned_data.csv')
df_cancelled = pd.read_csv('../data/cleaned/cancelled_transactions.csv')
df_returns = pd.read_csv('../data/cleaned/return_transactions.csv')

print(f"✅ df shape: {df.shape}")
print(f"✅ df_cancelled shape: {df_cancelled.shape}")
print(f"✅ df_returns shape: {df_returns.shape}")

# Phase 4 - SQL Exploration & Querying
Untuk fase ini kita akan membuat notebook sendiri, yaitu `02_sql_exploration.ipynb`. Ringkasan langkah dan output akan diberikan disini.

## Tujuan
Menjawab business questions awal menggunakan SQL query langsung ke `retail.db`.

## Query yang Dilakukan
1. **Total Revenue per Negara** → UK mendominasi (£14.389.234), diikuti EIRE & Netherlands
2. **Monthly Sales Trend** → Peak season Oktober–November, drop di Januari & Desember (data tidak full)
3. **Top 10 Produk Terlaris** → Gift & dekorasi rumah mendominasi; PAPER CRAFT hanya 1 order bulk
4. **Frekuensi Transaksi per Customer** → Top customers adalah wholesaler besar dengan AOV sangat tinggi
5. **Cancellation Rate per Negara** → Negara dengan volume kecil cenderung memiliki rate tinggi; Japan paling perlu diperhatikan (69.70%, 33 orders)
6. **Revenue by Month & Year (MoM Growth)** → Growth konsisten di Q4, Desember anomali karena data tidak full

## Catatan Teknis
- Query Cancellation Rate awalnya menggunakan `LEFT JOIN` yang menghasilkan logika salah
- Diperbaiki menggunakan `UNION ALL` untuk menghindari duplikasi baris antar tabel `retail` dan `cancelled`

## KPI Summary
| KPI | Nilai |
|---|---|
| Total Revenue | £17.374.804 |
| Total Orders | 36.969 |
| Total Customers | 5.878 |
| Total Products | 4.631 |
| Total Countries | 41 |
| AOV | £469.98 |

## Output Files
- `data/cleaned/kpi_summary.csv`
- `data/cleaned/monthly_trend.csv`

In [ ]:
# Load hasil phase 4

kpi = pd.read_csv('../data/cleaned/kpi_summary.csv')
monthly = pd.read_csv('../data/cleaned/monthly_trend.csv')

print("=== KPI SUMMARY ===")
print(f"Total Revenue    : £{kpi['Total_Revenue'].values[0]:,.2f}")
print(f"Total Orders     : {kpi['Total_Orders'].values[0]:,}")
print(f"Total Customers  : {kpi['Total_Customers'].values[0]:,}")
print(f"Total Products   : {kpi['Total_Products'].values[0]:,}")
print(f"Total Countries  : {kpi['Total_Countries'].values[0]:,}")
print(f"AOV              : £{kpi['AOV'].values[0]:,.2f}")

print(f"\n✅ Monthly trend loaded: {monthly.shape[0]} bulan")

# Phase 5 - Exploratory Data Analysis
Di phase ini kita akan membuat notebook baru `03_eda.ipynb` dan melakukan visualisasi untuk memahami data secara lebih mendalam.

## Tujuan
Memahami data secara visual melalui analisis univariate, bivariate, dan time series.

## Analisis yang Dilakukan
1. **Distribusi Revenue per Transaksi** → Right-skewed ekstrem; 95th percentile hanya £67.50 - mengindikasikan mayoritas transaksi bernilai kecil dengan segelintir wholesaler besar sebagai outlier
2. **Top 10 Produk Terlaris** → Didominasi produk dekorasi & gift; WORLD WAR 2 GLIDERS teratas (105.185 unit)
3. **Distribusi Pelanggan per Negara** → UK sangat dominan (5.350 customers); Germany di posisi 2 hanya 107 customers
4. **Monthly Revenue Trend** → Pola peak season Oct–Nov konsisten di 2010 & 2011; drop Desember karena data tidak full
5. **Revenue by Country** → UK (£14.389.235) mendominasi; EIRE & Netherlands unggul revenue meski jumlah customer kecil - mengkonfirmasi profil wholesaler
6. **Revenue by Day of Week** → Thursday tertinggi (£3.745.783); Saturday hampir nol (£9.803) - perusahaan kemungkinan tidak beroperasi di hari Sabtu
7. **Peak Season Identification** → November adalah bulan terkuat (£2.322.666), diikuti Oktober (£2.068.754)

## Key Findings
- Distribusi TotalPrice sangat right-skewed → perlu StandardScaler sebelum clustering di Phase 6
- Saturday hampir tidak ada transaksi → kemungkinan hari non-operasional
- Peak season Q4 (Oct–Nov) konsisten setiap tahun → peluang kampanye marketing musiman
- EIRE & Netherlands memiliki revenue tinggi dengan customer sedikit → segmen wholesaler paling valuable di luar UK

## Output Files
- `reports/01_revenue_distribution.png`
- `reports/02_top_products.png`
- `reports/03_customers_by_country.png`
- `reports/04_monthly_revenue.png`
- `reports/05_revenue_by_country.png`
- `reports/06_revenue_by_dow.png`
- `reports/07_peak_season.png`
- `data/cleaned/top_products.csv`
- `data/cleaned/revenue_by_country.csv`

In [ ]:
import matplotlib.pyplot as plt
import matplotlib.image as mpimg

# Load output data
top_products = pd.read_csv('../data/cleaned/top_products.csv')
revenue_country = pd.read_csv('../data/cleaned/revenue_by_country.csv')

print("=== EDA SUMMARY ===")
print(f"\nTop 3 Produk Terlaris:")
for i, row in top_products.head(3).iterrows():
    print(f"  {i+1}. {row['Description']} ({row['Quantity']:,} units)")

print(f"\nTop 3 Negara by Revenue:")
for i, row in revenue_country.head(3).iterrows():
    print(f"  {i+1}. {row['Country']} (£{row['Total_Revenue']:,.2f})")

print(f"\nPeak Season: Oktober & November")
print(f"Hari Teraktif: Thursday")
print(f"Hari Non-Operasional: Saturday")

# Tampilkan key visualisasi
fig, axes = plt.subplots(1, 2, figsize=(16, 5))
for ax, path, title in zip(axes,
    ['../reports/04_monthly_revenue.png', '../reports/07_peak_season.png'],
    ['Monthly Revenue Trend', 'Peak Season Identification']):
    img = mpimg.imread(path)
    ax.imshow(img)
    ax.axis('off')
    ax.set_title(title, fontweight='bold')
plt.tight_layout()
plt.show()

# Phase 6 - Customer Analytics
Phase ini adalah inti dari sisi analitik pelanggan. Terdiri dari 3 analisis utama:
- 6A. RFM Analysis
- 6B. K-Means Clustering
- 6C. Cohort Analysis

Kita akan buat file `04_customer_analytics.ipynb` dan ringkasannya sebagai berikut.

## Tujuan
Memahami perilaku pelanggan secara mendalam melalui tiga pendekatan analitik:
RFM Analysis, K-Means Clustering, dan Cohort Analysis.

---

## 6A - RFM Analysis

### Metodologi
- Recency: hari sejak transaksi terakhir (reference date: 2011-12-10)
- Frequency: jumlah invoice unik per customer
- Monetary: total revenue per customer
- Scoring: quintile 1–5 per dimensi (5 = terbaik)

### Segmen yang Dihasilkan
| Segmen | Jumlah | Avg Recency | Avg Frequency | Avg Monetary |
|---|---|---|---|---|
| Champions | 1.297 | 19.96 hari | 17.11x | £9.143 |
| Loyal Customers | 1.138 | 73.38 hari | 5.95x | £2.255 |
| Potential Loyalists | 358 | 79.72 hari | 2.70x | £591 |
| New Customers | 443 | 28.11 hari | 1.46x | £885 |
| Need Attention | 702 | 343.74 hari | 1.60x | £437 |
| Hibernating | 811 | - | - | - |
| At Risk | 616 | 359.51 hari | 5.62x | £4.244 |
| Lost | 513 | 568.23 hari | 1.00x | £344 |

### Key Insights
- **Champions (22.1%)** mendominasi revenue dengan Avg Monetary £9.143 -
  hampir 3x lipat segmen Loyal Customers
- **At Risk (10.5%)** adalah prioritas retensi utama - pernah bertransaksi
  besar (£4.244) tapi sudah 359 hari tidak aktif
- **Need Attention + Hibernating + Lost = 34.3%** dari total pelanggan -
  lebih dari sepertiga pelanggan sudah tidak aktif
- **New Customers (7.5%)** memiliki Avg Monetary £885 meski baru bergabung -
  potensi untuk di-nurture menjadi Loyal Customers

---

## 6B - K-Means Clustering

### Metodologi
- Fitur: Recency, Frequency, Monetary
- Preprocessing: Log Transformation (np.log1p) + StandardScaler
- Alasan Log Transformation: distribusi RFM sangat right-skewed,
  outlier wholesaler ekstrem dapat mendistorsi hasil clustering
- Jumlah cluster optimal: k=4 berdasarkan Elbow Method
- Variance explained oleh PCA: PC1 76.4% + PC2 18.7% = 95.1%

### Profil Cluster
| Cluster | Jumlah | Avg Recency | Avg Frequency | Avg Monetary |
|---|---|---|---|---|
| High Value | 1.196 | 27.71 hari | 19.28x | £10.731 |
| Promising | 1.250 | 28.30 hari | 3.05x | £857 |
| At Risk | 1.459 | 230.07 hari | 5.06x | £1.948 |
| Inactive | 1.973 | 394.95 hari | 1.38x | £317 |

### Key Insights
- **High Value & Promising memiliki Recency hampir sama** (27–28 hari) -
  yang membedakan adalah Frequency (19.3x vs 3.0x) dan
  Monetary (£10.731 vs £857), menunjukkan clustering berhasil
  memisahkan dua profil aktif yang berbeda
- **Inactive adalah cluster terbesar (1.973 customers)** - lebih dari
  33% pelanggan sudah tidak aktif lebih dari 1 tahun
- **At Risk masih memiliki nilai historis yang signifikan (£1.948)** -
  layak untuk diprioritaskan dalam win-back campaign
- **PCA menjelaskan 95.1% variance** - visualisasi 2D sudah sangat
  representatif untuk menggambarkan seluruh data

---

## 6C - Cohort Analysis

### Metodologi
- Pelanggan dikelompokkan berdasarkan bulan pertama bertransaksi
- Retention rate = % pelanggan yang kembali bertransaksi di bulan ke-N
- Total cohort: 25 bulan (Desember 2009 - Desember 2011)

### Key Insights
- **Retention bulan ke-1 berkisar 15–35%** - artinya 65–85% pelanggan
  tidak kembali di bulan berikutnya, umum untuk bisnis wholesale online
- **Cohort Desember 2009 paling loyal** - retention stabil di 25–45%
  hingga bulan ke-24, mengindikasikan pelanggan inti yang established
- **Peak season effect terlihat di diagonal heatmap** - nilai retention
  melonjak di bulan Oktober–November lintas cohort, membuktikan
  peak season mendorong returning customers bukan hanya new customers
- **Cohort Desember = one-time buyers** - retention bulan ke-1 hanya
  9.2% (2010-12), pelanggan holiday season cenderung tidak kembali
- **Mid-year cohort (Juli–Agustus) lebih loyal** dibanding cohort
  peak season - kemungkinan wholesaler yang mencari supplier rutin
- **Q3–Q4 2011 menunjukkan peningkatan retention** - kualitas akuisisi
  pelanggan baru yang semakin membaik menjelang akhir periode dataset

---

## Rekomendasi Bisnis

| Temuan | Segmen Target | Rekomendasi |
|---|---|---|
| At Risk punya nilai historis tinggi | At Risk (RFM & Cluster) | Win-back campaign dengan penawaran eksklusif |
| Champions = 22% customers, revenue terbesar | Champions | Loyalty program & early access produk baru |
| Desember = one-time buyers | New Customers (cohort Des) | Email onboarding series bulan Januari |
| Peak season mendorong returning customers | Semua segmen | Re-engagement campaign mulai September |
| Mid-year cohort paling loyal | - | Fokus akuisisi pelanggan di Q2–Q3 |
| Hibernating masih bisa di-recover | Hibernating | Reactivation campaign dengan diskon |

---

## Output Files
- `data/cleaned/rfm_segments.csv` ← RFM score, segmen & cluster label
- `data/cleaned/cluster_profile.csv` ← profil rata-rata per cluster
- `data/cleaned/cohort_retention.csv` ← retention matrix
- `reports/08_rfm_segments.png`
- `reports/09_elbow_method.png`
- `reports/10_cluster_pca.png`
- `reports/11_cluster_profile.png`
- `reports/12_cohort_heatmap.png`

In [ ]:
# Load hasil Phase 6
rfm = pd.read_csv('../data/cleaned/rfm_segments.csv')
cluster_profile = pd.read_csv('../data/cleaned/cluster_profile.csv')
cohort = pd.read_csv('../data/cleaned/cohort_retention.csv', index_col=0)

print("=== PHASE 6 SUMMARY ===")
print(f"\n📊 RFM Segmentation:")
for segment, count in rfm['Segment'].value_counts().items():
    pct = count / len(rfm) * 100
    print(f"  {segment:<25} : {count:>5} customers ({pct:.1f}%)")

print(f"\n🎯 K-Means Clustering:")
for _, row in cluster_profile.sort_values('Avg_Monetary', ascending=False).iterrows():
    print(f"  {row['Cluster_Label']:<15} : "
          f"R={row['Avg_Recency']:.0f}d | "
          f"F={row['Avg_Frequency']:.1f}x | "
          f"M=£{row['Avg_Monetary']:,.0f}")

print(f"\n📅 Cohort Analysis:")
print(f"  Total cohorts       : {len(cohort)} bulan")
print(f"  Avg retention M+1   : {cohort.iloc[:, 1].mean():.1f}%")
print(f"  Best cohort         : 2009-12 (retention stabil 25–45%)")
print(f"  Peak season effect  : Spike retention di Oct–Nov lintas cohort")

# Tampilkan key visualisasi
fig, axes = plt.subplots(1, 3, figsize=(18, 5))
for ax, path, title in zip(axes,
    ['../reports/08_rfm_segments.png',
     '../reports/10_cluster_pca.png',
     '../reports/12_cohort_heatmap.png'],
    ['RFM Segmentation',
     'K-Means Clustering (PCA)',
     'Cohort Retention Heatmap']):
    img = mpimg.imread(path)
    ax.imshow(img)
    ax.axis('off')
    ax.set_title(title, fontweight='bold')
plt.tight_layout()
plt.show()

# Phase 7 - Sales Analytics

Phase 7 terdiri dari 3 bagian utama:
- 7A. Revenue Analysis
- 7B. Product Analysis
- 7C. Cancellation Analysis

## Tujuan
Menganalisis performa penjualan dari tiga sudut pandang: Revenue Analysis,
Product Analysis, dan Cancellation Analysis untuk menjawab pertanyaan bisnis:
"Bagaimana tren performa penjualan kami, dan di mana peluang untuk meningkatkannya?"

---

## 7A - Revenue Analysis

### Key Insights
- **Total Revenue £17.374.804** dari 36.969 orders, 5.878 customers, dan
  4.631 produk dengan AOV £469.98
- **Pola seasonal sangat konsisten di kedua tahun** - revenue selalu
  memuncak di Q4 (Oktober–November) dan drop di awal tahun (Januari–Februari)
- **2010 Q4 (£2.769.995) vs 2011 Q4 (£2.709.056)** - meskipun terlihat
  sedikit turun, data 2011 Q4 tidak lengkap (hanya sampai pertengahan
  Desember), artinya 2011 Q4 sebenarnya lebih besar jika data penuh
- **2011 Q1 (£1.608.268)** adalah satu-satunya kuartal yang turun YoY
  dibanding 2010 Q1 (£1.757.340)
- **AOV stabil di kisaran £400–£575** sepanjang periode - tidak ada
  perubahan signifikan dalam nilai per transaksi
- **Tren AOV H2 2011 meningkat** - nilai per transaksi semakin besar
  menjelang akhir periode, mengindikasikan pelanggan memesan dalam
  jumlah yang lebih besar

---

## 7B - Product Analysis

### Key Insights
- **REGENCY CAKESTAND 3 TIER (£277.656)** adalah produk revenue tertinggi
  dengan 3.317 orders - jauh di atas produk kedua WHITE HANGING HEART
  T-LIGHT HOLDER (£247.048)
- **PAPER CRAFT, LITTLE BIRDIE (£168.470, 1 order)** - single bulk order
  dari satu wholesaler, bukan indikator performa produk yang sebenarnya
- **Top 21.7% produk menghasilkan 80% revenue** - mendekati prinsip
  Pareto 80/20, portofolio produk sangat terkonsentrasi
- **~78% produk (4.100+ produk) hanya berkontribusi 20% revenue** -
  kandidat untuk evaluasi dan rasionalisasi portofolio produk
- **Bottom 10 produk memiliki revenue di bawah £25** meski sudah
  memiliki 10+ orders - harga per unit sangat rendah, perlu evaluasi
  apakah masih layak dipertahankan

---

## 7C - Cancellation Analysis

### Key Insights
- **Cancellation rate 18.32% (orders)** - hampir 1 dari 5 orders
  dibatalkan, namun hanya **8.08% dari total revenue** - cancellation
  didominasi order bernilai kecil, bukan order besar
- **2.572 customers (43% dari total)** pernah melakukan cancellation
  setidaknya sekali
- **Tren cancellation rate menurun dari 2010 → 2011** - dari kisaran
  20–23% turun ke 14–17% di akhir 2011, mengindikasikan perbaikan
  operasional yang signifikan
- **Japan (41.1%)** memiliki cancellation rate tertinggi - lebih dari
  dua kali rata-rata (18.3%), kemungkinan ada masalah pengiriman
  internasional atau ekspektasi produk yang tidak terpenuhi
- **Netherlands (8.4%)** adalah negara dengan cancellation rate terendah
  - pelanggan paling konsisten dan reliable
- **11 dari 18 negara berada di bawah weighted average (18.3%)** -
  masalah cancellation terkonsentrasi di segelintir negara saja
- **REGENCY CAKESTAND 3 TIER (341 cancellations)** adalah produk
  paling sering dibatalkan sekaligus produk revenue tertinggi -
  mengindikasikan masalah inventory management pada produk high-demand

---

## Overall Insights - Phase 7

### Kekuatan Bisnis
- Revenue tumbuh secara konsisten dengan pola seasonal yang dapat
  diprediksi - memudahkan perencanaan inventory dan marketing
- AOV stabil dan cenderung meningkat di 2011 - pelanggan semakin
  nyaman melakukan pembelian dalam jumlah besar
- Cancellation rate (revenue) hanya 8.08% - dampak finansial
  cancellation masih terkendali
- Tren cancellation rate menurun - operasional bisnis semakin membaik

### Area yang Perlu Perhatian
- **Inventory Management** - produk terlaris (REGENCY CAKESTAND,
  WHITE HANGING HEART) sekaligus paling sering dibatalkan,
  mengindikasikan stockout pada produk high-demand
- **Konsentrasi Produk** - 78% produk hanya menghasilkan 20% revenue,
  portofolio terlalu lebar dan perlu rasionalisasi
- **Japan & Channel Islands** - cancellation rate jauh di atas
  rata-rata, perlu investigasi dan penanganan khusus untuk
  pasar internasional ini
- **Seasonal Dependency** - ketergantungan tinggi pada Q4 membuat
  bisnis rentan jika peak season tidak berjalan optimal

### Rekomendasi Bisnis
| Temuan | Rekomendasi |
|---|---|
| Produk terlaris = paling sering dicancel | Prioritaskan safety stock untuk top 10 produk |
| 78% produk hanya 20% revenue | Evaluasi & discontinue produk bottom performer |
| Japan cancel rate 41.1% | Investigasi pengalaman pelanggan Jepang |
| Peak season Q4 sangat dominan | Mulai pre-season campaign di Agustus–September |
| AOV meningkat di H2 2011 | Dorong bundle purchasing untuk tingkatkan AOV |
| Cancellation rate menurun | Pertahankan perbaikan operasional yang sudah berjalan |

---

## Output Files
- `data/cleaned/monthly_revenue.csv`
- `data/cleaned/quarterly_revenue.csv`
- `data/cleaned/kpi_summary.csv`
- `data/cleaned/product_analysis.csv`
- `data/cleaned/monthly_cancellation.csv`
- `data/cleaned/cancellation_by_country.csv`
- `data/cleaned/cancelled_products.csv`
- `reports/13_revenue_trend.png`
- `reports/14_revenue_by_quarter.png`
- `reports/15_aov_trend.png`
- `reports/16_product_performance.png`
- `reports/17_pareto_analysis.png`
- `reports/18_cancellation_trend.png`
- `reports/19_cancellation_by_country.png`
- `reports/20_top_cancelled_products.png`

In [ ]:
# ── Load semua output Phase 7 ──────────────────────────────────────────
monthly_revenue   = pd.read_csv('../data/cleaned/monthly_revenue.csv')
quarterly_revenue = pd.read_csv('../data/cleaned/quarterly_revenue.csv')
kpi_summary       = pd.read_csv('../data/cleaned/kpi_summary.csv')
product_analysis  = pd.read_csv('../data/cleaned/product_analysis.csv')
monthly_cancel    = pd.read_csv('../data/cleaned/monthly_cancellation.csv')
country_cancel    = pd.read_csv('../data/cleaned/cancellation_by_country.csv')
cancelled_products = pd.read_csv('../data/cleaned/cancelled_products.csv')

# ── Print ringkasan ────────────────────────────────────────────────────
print("=" * 50)
print("PHASE 7 - SALES ANALYTICS SUMMARY")
print("=" * 50)

print("\n📊 7A - Revenue Analysis:")
print(f"  Total Revenue      : £{kpi_summary['Total_Revenue'].values[0]:,.2f}")
print(f"  Total Orders       : {kpi_summary['Total_Orders'].values[0]:,}")
print(f"  Total Customers    : {kpi_summary['Total_Customers'].values[0]:,}")
print(f"  AOV                : £{kpi_summary['AOV'].values[0]:,.2f}")
best_month = monthly_revenue.loc[monthly_revenue['Monthly_Revenue'].idxmax()]
print(f"  Best Month         : {best_month['YearMonth']} "
      f"(£{best_month['Monthly_Revenue']:,.0f})")
best_quarter = quarterly_revenue.loc[quarterly_revenue['Quarterly_Revenue'].idxmax()]
print(f"  Best Quarter       : {best_quarter['YearQuarter']} "
      f"(£{best_quarter['Quarterly_Revenue']:,.0f})")

print("\n🛍️  7B - Product Analysis:")
print(f"  Total Products     : {len(product_analysis):,}")
top1 = product_analysis.iloc[0]
print(f"  Top Product        : {top1['Description']} "
      f"(£{top1['Total_Revenue']:,.0f})")
print(f"  Pareto 80% Revenue : Top 21.7% produk")
print(f"  Bottom products    : £{product_analysis.tail(10)['Total_Revenue'].max():,.2f} "
      f"max revenue (min 10 orders)")

print("\n❌ 7C - Cancellation Analysis:")
total_cancel_orders = country_cancel['Cancelled_Orders'].sum()
total_all_orders    = country_cancel['Total_Orders'].sum()
cancel_rate         = total_cancel_orders / total_all_orders * 100
cancel_revenue      = monthly_cancel['Cancelled_Revenue'].sum()
print(f"  Cancelled Orders   : {total_cancel_orders:,}")
print(f"  Cancellation Rate  : {cancel_rate:.2f}%")
print(f"  Cancelled Revenue  : £{cancel_revenue:,.2f}")
top_cancel_country  = country_cancel.iloc[0]
print(f"  Highest Cancel Rate: {top_cancel_country['Country']} "
      f"({top_cancel_country['Cancel_Rate']:.1f}%)")
top_cancel_product  = cancelled_products.iloc[0]
print(f"  Most Cancelled     : {top_cancel_product['Description']} "
      f"({top_cancel_product['Cancelled_Orders']:,.0f} orders)")

# ── Tampilkan visualisasi key Phase 7 ─────────────────────────────────
fig, axes = plt.subplots(2, 3, figsize=(20, 10))

visuals = [
    ('../reports/13_revenue_trend.png',        'Revenue Trend & MoM Growth'),
    ('../reports/14_revenue_by_quarter.png',   'Revenue by Quarter'),
    ('../reports/15_aov_trend.png',            'AOV Trend'),
    ('../reports/16_product_performance.png',  'Product Performance'),
    ('../reports/18_cancellation_trend.png',   'Cancellation Trend'),
    ('../reports/19_cancellation_by_country.png', 'Cancellation by Country'),
]

for ax, (path, title) in zip(axes.flatten(), visuals):
    img = mpimg.imread(path)
    ax.imshow(img)
    ax.axis('off')
    ax.set_title(title, fontweight='bold', fontsize=10)

plt.suptitle('Phase 7 - Sales Analytics Overview', 
             fontsize=14, fontweight='bold')
plt.tight_layout()
plt.show()

# Phase 8 - Dashboard Development